server/service/apple_mdm.go

```diff
@@ -6992,6 +7110,18 @@ func (svc *Service) MDMAppleProcessOTAEnrollment(
 		return nil, ctxerr.Wrap(ctx, err, "creating new host record")
 	}
 
+	if idpUUID != "" {
+		// TODO(mna): associate the host with the IdP UUID account if provided.
+		ctx = ctxdb.RequirePrimary(ctx, true)
+		h, err := svc.ds.HostLiteByIdentifier(ctx, deviceInfo.Serial)
+		if err != nil {
+			return nil, ctxerr.Wrap(ctx, err, "load host record")
+		}
+		if err := svc.ds.AssociateHostMDMIdPAccount(ctx, h.UUID, idpUUID); err != nil {
+			return nil, ctxerr.Wrap(ctx, err, "associate host with idp account")
```

Comment:
Maybe we should log+ignore the error if the `idpUUID` does not match an existing account.